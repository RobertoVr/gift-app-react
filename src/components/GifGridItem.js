import React from 'react'

export const GifGridItem = ({ id, title, url }) => {
    return (
        <div>
            <p>{title}</p>
            <img src={url} alt="gif" />
        </div>
    )
}
