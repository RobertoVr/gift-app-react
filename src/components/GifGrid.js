import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    useEffect(() => {
        getGifs();
    }, []);

    const getGifs = async () => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Goku&limit=10&api_key=PVTSe8E8rAfl36W4IWV3d2GoSMEkLoVh';
        const respuesta = await fetch(url);
        const { data } = await respuesta.json();
        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        setImages(gifs);
    }

    return (
        <div>
            {
                images.map((img) => (
                    <GifGridItem
                        key={img.id}
                        {...img}
                    />
                ))
            }
        </div>
    )
}
