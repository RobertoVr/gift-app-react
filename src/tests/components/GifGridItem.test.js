import React from 'react'
import '@testing-library/jest-dom'
import { GifGridItem } from '../../components/GifGridItem';
import { shallow } from 'enzyme';

describe('Pruebas sobre gifgrid', () => {

    const title = 'Un titulo';
    const url = 'https://test.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('Renderizar GifGridItem', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('Debe tener un parrafo con el titulo', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })
    test('Debe tener un src y alt la img', () => {
        const img = wrapper.find('img');
        // console.log(img.props().src)
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })
    test('Debe tener animate__fadeIn', () => {
        const div = wrapper.find('div');
        const result = div.props().className.includes('animate__fadeIn');
        // console.log(div.props().className)
        expect(result).toBe(true);
    })


});
