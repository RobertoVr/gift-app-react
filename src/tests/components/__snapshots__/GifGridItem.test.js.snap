// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre gifgrid Renderizar GifGridItem 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <p>
    Un titulo
  </p>
  <img
    alt="Un titulo"
    src="https://test.jpg"
  />
</div>
`;
